server/src/main/java/org/candlepin/pki/impl/BouncyCastleProviderLoader.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.pki.impl;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import java.security.Security;
+
+/** When this class is loaded, the Bouncy Castle provider will be installed into the JVM.  The
+ * provider can also be added via a call to a static method if needed in a test for example.
```

Comment:
Please consider mentioning in this comment that `addProvider` can be called more than once with the same provider with no issue (from [the javadocs](https://docs.oracle.com/javase/7/docs/api/java/security/Security.html#addProvider(java.security.Provider)), it will just return -1).

When I first looked I wondered if multiple calls might cause issues...

I think ideally we'd ensure our code only calls `addProvider` once, at initialization, where initialization can be servlet context creation, unit test loading, etc. but after reading the `java.security.Provider.addProvider` javadoc, I think it's no big deal to call it a few places (as is currently written).

Reply:
Done.

I would rather not squash in that "fix random stuff" commit because a lot of it is not logically related to a switch to BouncyCastle.  Like there are changes in that commit to the Huffman encoding process that it independent of the crypto provider.  I just made the changes to get rid of a lot of warnings before I started the migration in earnest.